pkg/operation/botanist/migration.go

```diff
@@ -102,43 +102,40 @@ func (b *Botanist) restoreExtensionObject(ctx context.Context, client client.Cli
 }
 
 func (b *Botanist) restoreExtensionObjectState(ctx context.Context, client client.Client, extensionObj extensionsv1alpha1.Object, resourceKind, resourceName string, purpose *string) error {
-	return kutil.TryUpdateStatus(ctx, k8sretry.DefaultBackoff, client, extensionObj, func() error {
-		var resourceRefs []autoscalingv1.CrossVersionObjectReference
-		if b.ShootState.Spec.Extensions != nil {
-			list := gardencorev1alpha1helper.ExtensionResourceStateList(b.ShootState.Spec.Extensions)
-			if extensionResourceState := list.Get(resourceKind, &resourceName, purpose); extensionResourceState != nil {
-				extensionStatus := extensionObj.GetExtensionStatus()
-				extensionStatus.SetState(*extensionResourceState.State)
-				extensionStatus.SetResources(extensionResourceState.Resources)
-				for _, r := range extensionResourceState.Resources {
-					resourceRefs = append(resourceRefs, r.ResourceRef)
-				}
+	var resourceRefs []autoscalingv1.CrossVersionObjectReference
+	if b.ShootState.Spec.Extensions != nil {
+		list := gardencorev1alpha1helper.ExtensionResourceStateList(b.ShootState.Spec.Extensions)
+		if extensionResourceState := list.Get(resourceKind, &resourceName, purpose); extensionResourceState != nil {
+			extensionStatus := extensionObj.GetExtensionStatus()
+			extensionStatus.SetState(*extensionResourceState.State)
```

Comment:
We could probably avoid passing the State and modify `SetState` to take a pointer now that we know the true cause of the error. Wdyt?